### PR TITLE
TS-3726: Expose build_error_response

### PR DESCRIPTION
**Pull request by oknet**

### CODE REVIEW

File: proxy/api/ts/ts.h [hunk 1]
```diff
@@ -2336,6 +2336,35 @@ tsapi uint64_t TSHttpTxnIdGet(TSHttpTxn txnp);
...
+          is NULL, the default HTTP reason phrase is used.
+   @param error_body_type the error message type, as specified by the HttpBodyFactory
+          customized error page system. Get the list by command:
+             "ls -al etc/trafficserver/body_factory/default/"
+
+   @param format If it is not NULL or "", it is also added to the error text 
+          body as descriptive text in the error body.
+
+   @note reference from build_error_response: 
+         This routine DOES NOT check for buffer overflows.  The caller should keep
```
Comment (alhonen) [inline proxy/api/ts/ts.h]: Do we really want to expose build_error_response() as long as this is the case?

Comment (zwoop) [inline proxy/api/ts/ts.h]: I agree. My thoughts was that we'd first cleanup the underlying body factory code (separate Jira's for that), such that we get clean and efficient APIs. And then we add public plugin APIs on top of that

So, I agree that for now, we probably shouldn't commit this, but once you (Jari) are done with the body factory changes, you can perhaps work with Oknet on getting this merged?

I hope that make sense?

**Comment by zwoop** [comment]
@oknet  For now, I think we should close this PR, wait for @alhonen to clean up the underlying plumbing, and then revisit this? If you agree, please close this PR for, but obviously keep your changes for future.

**Comment by oknet** [comment]
@zwoop ok, close this PR and wait for @alhonen.

**Pull request closed without merging**